src/map/Map.js

```diff
@@ -1625,7 +1630,20 @@ export var Map = Evented.extend({
 			this._animateToCenter = center;
 			this._animateToZoom = zoom;
 
+			if (duration) {
+				if (typeof (duration) === 'number') {
+					duration = duration + 's';
+				}
+				this._zoomStyle = document.createElement('style');
+				this._zoomStyle.innerHTML = '.leaflet-zoom-anim .leaflet-zoom-animated {' +

```

Comment:
Nope nope nope nope. This will mess up things when there are several map instances in the same page zooming at the same time. This should rather be done by setting `this._container.style.transition` (or rather, `this._container.style[DomUtil.TRANSITION]` for compatibility).

Also, with the current implementation new `<style>` elements would be added to the map pane, but never removed (even after their contents are set to an empty string on the timeout).